### DISCUSSION TITLE
Some features assume keyboard... No go on phones

### DISCUSSION THREAD

**Discussion by programmin1** [ezra-bible-app/ezra-bible-app#658] [General]
I noticed the dictionary feature instructs to shift+mouse over the word.

That's not going to work on Librem 5 and other Linux phones while not docked to some sort of screen+keyboard setup.

**Comment by tobias-klein** [comment]
This is already working on Android via long-press. Just the help instruction in the dictionary panel still needs to be updated (see #623). We can re-use the [existing long-press approach](https://github.com/ezra-bible-app/ezra-bible-app/blob/master/app/frontend/controllers/dictionary_controller.js#L182), but need to reliably detect the platform. The detection methods are implemented in [platform_helper.js](https://github.com/ezra-bible-app/ezra-bible-app/blob/master/app/lib/platform_helper.js). Can you submit a pull request that adds a detection method for Librem?

**Comment by programmin1** [comment]
One could check that navigator.platform =='Linux aarch64'

However I imagine there are and will be laptops that have that arch. And Librem5 can be used as a laptop or desktop with a usbc dock.

How about checking window.innerWidth < 600 or so to detect phones?

The odd thing is this detects as a Nexus in the Electron inspector:
navigator.userAgent
'Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.109 Mobile Safari/537.36'

**Comment by tobias-klein** [comment]
The majority of phones would already be covered by the Cordova version of the app, if we enable small screens in the Android app configuration (currently disabled). For style topics we should be good with the css responsive design based on screen dimensions.
Specifically regarding the touch interface, we could also decide to support touch long-press and mouse use at the same time - that would be needed for example on Windows tablets (Surface family) as well.
I'll look into this and see what we can do.
